danger_code_ws/2.cpp

```diff
@@ -0,0 +1,87 @@
+#include <iostream>
+#include <cstdlib>
+
+// Function to dynamically allocate a matrix of size rows x cols
+double** allocateMatrix(int rows, int cols) {
+    double** matrix = new double*[rows];
+    for (int i = 0; i < rows; i++) {
+        matrix[i] = new double[cols];
+    }
+    return matrix;
+}
+
+// Function to initialize a matrix with random values
+void initializeMatrix(double** matrix, int rows, int cols) {
+    for (int i = 0; i < rows; i++) {
+        for (int j = 0; j < cols; j++) {
+            matrix[i][j] = (std::rand() % 100) / 10.0;  // Random values between 0 and 10
```

Comment:
Using `std::rand()` for generating random numbers is generally discouraged for numerical applications. Consider using a more robust random number generator from `<random>`, such as `std::mt19937` with a suitable distribution (e.g., `std::uniform_real_distribution`). This provides better statistical properties and avoids potential biases.